src/dependency_specifications.rb

```diff
@@ -0,0 +1,31 @@
+
+# frozen_string_literal: true
+
+module Metalware
+  # Class to generate reused `dependency_hash`s, for use by `Dependency` object
+  # to enforce command dependencies.
+  # XXX Consider moving generating of more `dependency_hash`s here.
+  class DependencySpecifications
+    def initialize(config)
+      @config = config
+    end
+
+    def for_node_in_configured_group(node_name)
+      primary_group = configured_primary_group_for_node(node_name)
+      {
+        repo: ['configure.yaml'],
+        configure: ['domain.yaml', "groups/#{primary_group}.yaml"],
```

Comment:
OK - adding a distinct method for that sounds reasonable, otherwise it's going to make things quite a bit more complicated if we start having a lot of optionally added files in the dependencies.